session/session.go

```diff
@@ -517,7 +517,7 @@ func (s *session) doCommit(ctx context.Context) error {
 	s.txn.SetOption(tikvstore.EnableAsyncCommit, s.GetSessionVars().EnableAsyncCommit)
 	s.txn.SetOption(tikvstore.Enable1PC, s.GetSessionVars().Enable1PC)
 	// priority of the sysvar is lower than `start transaction with causal consistency only`
-	if s.txn.GetOption(tikvstore.GuaranteeLinearizability) == nil {
+	if val := s.txn.GetOption(tikvstore.GuaranteeLinearizability); val == nil || val.(bool) {
```

Comment:
`val == nil` seems impossible because `IsCasualConsistency` always returns a bool?

Reply:
There are some other mock store implementations. I guess this way is better as it requires less detail of underlying engine.